services/src/main/java/org/exoplatform/webconferencing/WebConferencingService.java

```diff
@@ -0,0 +1,1235 @@
+   * @param userId the user id
+   */
+  protected void fireUserCallJoined(String callId,
+                                    String providerType,
+                                    String ownerId,
+                                    String ownerType,
+                                    String partId,
+                                    String userId) {
+    Set<UserCallListener> listeners = userListeners.get(userId);
+    if (listeners != null) {
+      for (UserCallListener listener : listeners) {
+        if (listener.isListening()) {
+          listener.onPartJoined(callId, providerType, ownerId, ownerType, partId);
+        }
+      }
+    }
+  }
+
+  /**
+   * Fire user call part leaved.
+   *
+   * @param callId the call id
+   * @param providerType the provider type
+   * @param ownerId the owner id
+   * @param ownerType the owner type
+   * @param partId the part id
+   * @param userId the user id
+   */
+  protected void fireUserCallLeaved(String callId,
+                                    String providerType,
+                                    String ownerId,
+                                    String ownerType,
+                                    String partId,
+                                    String userId) {
+    Set<UserCallListener> listeners = userListeners.get(userId);
+    if (listeners != null) {
+      for (UserCallListener listener : listeners) {
+        if (listener.isListening()) {
+          listener.onPartLeaved(callId, providerType, ownerId, ownerType, partId);
+        }
+      }
+    }
+  }
+
+  /**
+   * Adds the plugin.
+   *
+   * @param plugin the plugin
+   */
+  public void addPlugin(ComponentPlugin plugin) {
+    Class<CallProvider> pclass = CallProvider.class;
+    if (pclass.isAssignableFrom(plugin.getClass())) {
+      addProvider(pclass.cast(plugin));
+    } else {
+      LOG.warn("Video Calls provider plugin is not an instance of " + pclass.getName() + ". Skipped plugin: " + plugin);
+    }
+  }
+
+  /**
+   * Adds the provider.
+   *
+   * @param provider the provider
+   */
+  public void addProvider(CallProvider provider) {
+    for (String type : provider.getSupportedTypes()) {
+      CallProvider existing = providers.putIfAbsent(type, provider);
+      if (existing != null) {
+        LOG.warn("Video Calls provider type '" + existing.getType() + "' already registered. Skipped plugin: " + provider);
+      }
+    }
+  }
+
+  /**
+   * Gets the provider.
+   *
+   * @param type the type
+   * @return the provider
+   */
+  public CallProvider getProvider(String type) {
+    return providers.get(type);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void start() {
+    // XXX SpaceService done in crappy way and we need reference it after the container start only, otherwise
+    // it will fail the whole server start due to not found JCR service
+    this.spaceService = ExoContainerContext.getCurrentContainer().getComponentInstanceOfType(SpaceService.class);
```

Comment:
There is no need to make this service Startable IMO. The SpaceService can be retrieved at its first access instead:
if(spaceService == null) {
  spaceService = ExoContainerContext.getCurrentContainer().getComponentInstanceOfType(SpaceService.class);
}
return spaceService;

Reply:
Check on null will happen each time we need space service then, but when initialized on the start, we just know it is here.